### ISSUE TITLE
您好，对任务一里用numpy实现的LogisticRegression有点疑问

### ISSUE THREAD

**Issue by YerayL** [Stark-JC/code-for-nlp-beginner#2]
范数正则化那里：
       if self.regularization == "l2":
            self.w += self.learning_rate * (gradients * self.C - self.w)
        elif self.regularization == "l1":
            self.w += self.learning_rate * (gradients * self.C - np.sign(self.w))
        else:
            self.w += self.learning_rate * gradients
l1和l2范数正则化时，正则化系数self.C为啥不是乘以右边的self.w和np.sign(self.w)，而是左边的gradients吧？

**Comment by YerayL** [comment]
还有两个问题也是LogisticRegression里面的，为什么要用_exp_dot，不直接dot？

**Comment by jiacheng-ye** [comment]
你好，
第一个问题，其实是等价的，看你怎么定义C，可参考：https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
第二个问题，可以看看《统计学习方法》p79的推导
